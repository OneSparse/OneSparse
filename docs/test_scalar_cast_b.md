# Scalar `bool` Casting

This documentation is also tests for the code, the examples below
show the literal output of these statements from Postgres.

Some setup to make sure warnings are shown, and that the extension
is installed.
``` postgres-console
set client_min_messages = 'WARNING';
create extension if not exists onesparse;
```
These casting functions cast the Postgres type `bool` to the
GraphBLAS scalar type `GrB_BOOL`.
``` postgres-console
select cast(0::bool as scalar);
 scalar 
--------
 b:f
(1 row)

select cast(0::bool as scalar);
 scalar 
--------
 b:f
(1 row)

select cast(1::bool as scalar);
 scalar 
--------
 b:t
(1 row)

```
These tests cast back from the scalar type `GrB_BOOL` to the
Postgres type `bool`
``` postgres-console
select cast(0::bool::scalar as bool);
 bool 
------
 f
(1 row)

select cast(0::bool::scalar as  bool);
 bool 
------
 f
(1 row)

select cast(1::bool::scalar as scalar);
 scalar 
--------
 b:t
(1 row)

```