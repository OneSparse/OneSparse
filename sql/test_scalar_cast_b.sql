set client_min_messages = 'WARNING';
create extension if not exists onesparse;
set search_path = 'onesparse';
select cast(0::bool as scalar);
select cast(0::bool as scalar);
select cast(1::bool as scalar);
select cast(0::bool::scalar as bool);
select cast(0::bool::scalar as  bool);
select cast(1::bool::scalar as scalar);
