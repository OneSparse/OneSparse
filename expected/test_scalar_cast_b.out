set client_min_messages = 'WARNING';
create extension if not exists onesparse;
set search_path = 'onesparse';
select cast(0::bool as scalar);
 scalar 
--------
 b:f
(1 row)

select cast(0::bool as scalar);
 scalar 
--------
 b:f
(1 row)

select cast(1::bool as scalar);
 scalar 
--------
 b:t
(1 row)

select cast(0::bool::scalar as bool);
 bool 
------
 f
(1 row)

select cast(0::bool::scalar as  bool);
 bool 
------
 f
(1 row)

select cast(1::bool::scalar as scalar);
 scalar 
--------
 b:t
(1 row)

